### ISSUE TITLE
Colors of ls

### ISSUE THREAD

**Issue by isadoramariamendes** [mikeizbicki/ucr-cs100#1463]
I created a function that changes the output color and one that resets the color to the normal... So after using the color change function, I reset, always.
It works fine in my machine, as it suppose to work... green for execs, blue for dirs, gray for hidden files... However, in hammer just dont print the green... :( I don't know why...
Here are some ss about this issue:

![No colors](http://oi61.tinypic.com/2q8y9g3.jpg)

![With color](http://oi58.tinypic.com/rvlens.jpg)

I am worried because it will appear in the script that the color "didn't work". But it is...

**Comment by ghost** [comment]
You should only really need to change the color of the text. Like to colors: "\x1b[34m" or "\x1b[32m"?
Try tracing your code ... like tell it cout a message to let you know that your code attempted to change the text colors. If it did, you might want to check the color you choose, it might not be supported.

**Comment by mikeizbicki** [comment]
@isadoramariamendes I don't understand your question.  You're showing screenshots of hammer where the color appears to be working just fine.  Can you rephrase it somehow?

**Comment by isadoramariamendes** [comment]
Mike,
I am sorry... One ss is from hammer and doesn't work properly... the other is my local machine, where it works... i.e., it works in the local machine, but not on hammer...

**Comment by mikeizbicki** [comment]
@isadoramariamendes My confusion is that I see color in both images.  I recognize that the top one is from hammer, but I don't see what's wrong with it.

**Comment by mikeizbicki** [comment]
Oh, I see now.  Everything but green works.

This means you're not actually using the ANSI color codes correctly and you will not get the extra credit without fixing it.

**Comment by isadoramariamendes** [comment]
I am using something like:

```
    if(background == 1) cout << "\x1b[47m";
    if(S_ISDIR(protec)) cout << "\x1b[34m";
    else if(protec & S_IXUSR) cout << "\x1b[32m";
```

That's a pretty simple code, actually, that's why it's so weird... For each print I call this function and after all prints I "reset" the color. Seems like in some computers it works and in some it doesn't... I don't know what else I can do... :(
